packages/itwin/tree-widget/src/test/trees/models-tree/ModelsTree.test.tsx

```diff
@@ -252,13 +254,36 @@ describe("ModelsTree", () => {
         }));
       });
 
+      it("should call visibilityHandler with correct parameters when it is passed as a function", async () => {
+        const node: TreeNodeItem = { id: "test", label: PropertyRecord.fromString("test-node"), isCheckboxVisible: true };
+        setupDataProvider([node]);
+        visibilityHandlerMock.getVisibilityStatus = async () => ({ state: "hidden" });
+        const getVisibilityStatusSpy = sinon.spy(visibilityHandlerMock, "getVisibilityStatus");
+        const visibilityHandlerCallback = (_: ModelsVisibilityHandlerProps) => {return visibilityHandlerMock;};
```

Comment:
This test is not doing what it says - it says it's verifying that the factory is called with correct props, but really it's checking if the handler's `getVisibilityStatus` function is called with correct props. We should spy on the factory instead.